tuning/sft_trainer.py

```diff
@@ -268,6 +277,22 @@ def train(
         peft_config=peft_config,
     )
 
+    # We track additional metrics and experiment metadata after
+    # Trainer object creation to ensure that this is not repeated
+    # multiple times for FSDP runs.
+    if trackers is not None and trainer.is_world_process_zero():
```

Comment:
@dushyantbehl @kmehant just have one question - since we are leaving it to user to pass tracker in train() and pass its associated callbacks as well, will any errors happen in case user passes aim tracker but doesnt pass aim callback? - its ok if errors are returned and user understands they need to correct arguments

I know in main() we extract callbacks from tracker and pass them to train(), but I just thinking if a user misses that step and calls train with incomplete set of callbacks, will it silently be ignored or would there be an indication to user?

Reply:
do you think we should move that code here to extract callbacks from a tracker in train function ? and in callbacks arguments, users should give any additional not specified by tracker? 
or do you want users to manually pass callbacks to train() that are needed for tracker as separate argument - is there a use case where users may not want to use callbacks of a tracker/ optionally select some?

Reply:
I did just that and updated the code.